CAIPs/caip-x.md

```diff
@@ -0,0 +1,110 @@
+
+
+**Request Provider Mechanism**:
+
+To ensure dApps can discover wallets even if they start listening after wallets have emitted their initial events, a request mechanism is essential.
+
+- **Active Request Capability**: Wallets must listen for a specific event from dApps that requests the announcement of available wallet providers. Upon receiving this event, wallets should re-announce their presence.
+
+- **Event Handling**: Wallets should listen for a `requestWalletExtensionId` event and respond by re-dispatching their provider information using the `walletExtensionIdAnnouncement` event. This ensures that any dApp can discover all available wallets, regardless of the script execution order.
+
+### Example Implementation
+
+- **Wallets**:
+
+```javascript
+function announceExtensionId() {
+  const providerInfo = {
+    name: 'Example Wallet',
+    icon: 'data:image/svg+xml,<svg xmlns="http://www.w3.org/2000/svg"/>',
+    rdns: 'com.example.wallet',
+    extensionId: 'your_extension_id_here'
+  };
+  document.dispatchEvent(new CustomEvent('walletExtensionIdAnnouncement', {
+    detail: Object.freeze(providerInfo)
+  }));
+}
+
+announceExtensionId();
+
+window.addEventListener('requestWalletExtensionId', () => {
+  announceExtensionId();
+});
+```
+
+- **dApps**:
+
+```javascript
+window.addEventListener('walletExtensionIdAnnouncement', function(e) {
+  const providerInfo = e.detail;
+  // Use providerInfo.extensionId to establish communication
+});
+
+window.dispatchEvent(new Event("requestWalletExtensionId"));
+```
+
+**Content Script Utilization**:
+While this CAIP significantly advances the way wallets interact with dApps through the `externally_connectable` API, it still necessitates the use of a content script. This script, however, requires fewer permissions compared to traditional methods and can be injected synchronously to facilitate the initial communication process. This adjustment is necessary to bridge the current browser limitations and set the stage for future enhancements.
+
+- **Synchronous Injection**: The content script can be injected synchronously with the page load, ensuring that the wallet's extension ID is available as soon as the dApp begins its initialization process. This method avoids the delays and potential conflicts associated with asynchronous script loading, leading to a smoother and more predictable interaction pattern.
+    
+- **Security and Efficiency**: Although using a content script still presents a minimal footprint on the user's browsing experience, this approach is a compromise that balances current browser capabilities with the need for enhanced security and performance. By minimizing the script's functionality to only what is necessary for ID transmission, the risks associated with broader access permissions are mitigated.
```

Comment:
thoughts on perhaps highlighting how this solution is imperfect but meant to be temporary. Instead it paves a path towards a future in which it can be superseded by a browser native solution? maybe in a different section? maybe this isn't necessary, idk